<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:macro name="bot_differential_drive_control" params="name prefix use_mock_hardware">

        <ros2_control name="${name}" type="system">
            <xacro:unless value="${use_mock_hardware}">
                <hardware>
                    <plugin>bot_differential_drive/DifferentialDriveSerialInterface</plugin>
                    <param name="left_wheel_name">left_wheel_joint</param>
                    <param name="right_wheel_name">right_wheel_joint</param>
                    <param name="loop_rate">1000</param>
                    <param name="device_name">/dev/ttyACM0</param>
                    <param name="baud_rate">9600</param>
                    <param name="timeout_ms">1000</param>
                </hardware>
            </xacro:unless>
            <!--
            <xacro:if value="${use_mock_hardware}">
                <hardware>
                    <plugin>bot_differential_drive/DifferentialDriveMock</plugin>
                    <param name="left_wheel_name">left_wheel_joint</param>
                    <param name="right_wheel_name">right_wheel_joint</param>
                    <param name="loop_rate">1000</param>
                    <param name="device_name">/dev/ttyACM0</param>
                    <param name="baud_rate">9600</param>
                    <param name="timeout_ms">1000</param>
                </hardware>
            </xacro:if>
            -->
            <xacro:if value="${use_mock_hardware}">
                <hardware>
                    <plugin>mock_components/GenericSystem</plugin>
                    <param name="calculate_dynamics">true</param>
                </hardware>
            </xacro:if>
            <joint name="${prefix}left_wheel_joint">
                <command_interface name="velocity"/>
                <state_interface name="position">
                    <param name="initial_value">0.0</param>
                </state_interface>
                <state_interface name="velocity">
                    <param name="initial_value">0.0</param>
                </state_interface>
            </joint>
            <joint name="${prefix}right_wheel_joint">
                <command_interface name="velocity"/>
                <state_interface name="position">
                    <param name="initial_value">0.0</param>
                </state_interface>
                <state_interface name="velocity">
                    <param name="initial_value">0.0</param>
                </state_interface>

            </joint>
        </ros2_control>
    </xacro:macro>

</robot>